### DISCUSSION TITLE
Radiant mod

### DISCUSSION THREAD

**Discussion by mk56-spn** [ppy/osu#16312] [Ideas]
(Whilst I have two other things in the works they are close to completion so I thought I'd open this idea up)

The idea here is to make a mod for the osu!std mode which simulates simple lighting effects and has them interact with the cursor

Each hit circle and sliderhead would have "bloom",as can be seen here (screenshot pulled from  "danser" https://github.com/Wieku/danser-go)

![IMG_20220103_203716.jpg](https://user-images.githubusercontent.com/74463310/147972946-a4e5bc71-fde3-4389-8c33-402936a3450b.jpg)

Except most likely considerably stronger and adjustable. 

The cursor would use the overlapping total opacity of the bloom of the hitcircles/sliderhead at its location  as it's color4 alpha level. This means it would be fully visible near the center of a hitcircle but invisible far away from any. Giving the effect of the hitcircles being light sources

Slider would be made invisible by default an  illuminated circle tracing the path ahead of the player at the same speed that the AR draws the hitcircles in.

Other options could potentially be included such as tying the intensity of a hitcircles light to the pulsing that's currently used for the osu logo

I do not intend to make this another gimmick mod nobody wants to play but an actually enjoyable difficulty increasing mod that people would play regularly and potentially even main. As such I'm open to suggestions of how this idea could be made more fun 

Ps. Approach circles will be hidden which is why I was considering pulsing the hitcircles to help keep players on beat

**Comment by peppy** [comment]
Mods generally aren't used to provide visual effects. This would be implemented as a skin or game-wide feature (similar to the shader effects stable has).

**Reply by mk56-spn** [reply]
The difference is these are difficulty affecting and more akin to HD or FL than a skin effect. Only the pulsing hitcircles could be considered skin material . The circle bloom and cursor visibility doesn't seem fitting no?

**Reply by peppy** [reply]
Visual effects do not affect difficulty.

**Reply by mk56-spn** [reply]
> Visual effects do not affect difficulty.

Id argue that the cursors visibility depending on its distance to the nearest circle very much affects the difficulty

**Reply by integerrr** [reply]
Is this not basically no-scope mod but easier then?

**Reply by mk56-spn** [reply]
> Is this not basically no-scope mod but easier then?

No this has more features. I guess it could be split into a no scope subsetting and have lighting effects in skinning. The point in grouping them is I wanted the lighting effects to be customizable in strength as a mod to determine at what point the cursor fades out . Which affects the difficulty alot. If it's moved to a skin element this cannot be done unless we allow mod pp to check skin settings which seems stupidly counter intuitive

**Reply by Wieku** [reply]
I think another argument here is that bloom is a fairly expensive operation as it needs a blur pass. So I guess game modifier requiring a stronger PC is a no-go as it's kinda a pay-to-win situation.

Here's the impact in danser for example on gtx 1080. And the impact would be bigger if not for the fact that bloom here is at a quarter the size, sacrificing fidelity a bit: https://img.wieku.me/22/2022-11-03%2013-40-56.mp4

**Reply by mk56-spn** [reply]
> I think another argument here is that bloom is a fairly expensive operation as it needs a blur pass. So I guess game modifier requiring a stronger PC is a no-go as it's kinda a pay-to-win situation.
> 
> Here's the impact in danser for example on gtx 1080. And the impact would be bigger if not for the fact that bloom here is at a quarter the size, sacrificing fidelity a bit: https://img.wieku.me/22/2022-11-03%2013-40-56.mp4

whilst i no longer think this mod is a good idea for the most part and dont wish to get it into the game, i dont think that performance is inherently a good arguement as lazer stands right now, considering theres a lot of optimisation to do from what i understand , take freeze frame, running on a 1070ti ( arch linux ) ryzen 2600x i drop down to 30fps from 960 at times ( whilst purposefully stress testing it with bleed

**Reply by Wieku** [reply]
But it's not about optimization. Bloom/blur require kinda a lot of raw GPU computing power, you can't optimize that without sacrificing quality.
